public/pages/DetectorResults/containers/AnomalyResults.tsx

```diff
@@ -243,6 +253,29 @@ export function AnomalyResults(props: AnomalyResultsProps) {
         ''
       );
     }
+    if (isPerformingColdStart) {
+      return (
+        <div>
+          <EuiFlexGroup direction="row" gutterSize="s">
+            <EuiLoadingSpinner
+              size="l"
+              style={{
+                marginLeft: '4px',
+                marginRight: '8px',
+                marginBottom: '8px',
+              }}
+            />
+            <EuiText>
+              <p>
+                Attempting to initialize the detector with historical data. This
+                will take approximately{' '}
+                {detector.detectionInterval.period.interval} total minutes.
```

Comment:
This is true if source index has enough data. How about the case that source index has no enough data ?

Reply:
Wouldn't the cold start process time be pretty independent of the amount of data in the source index? It either finds data for the model, or doesn't, but will still finish quick (I've never been able to make the cold start process last longer than a detector interval).

If there's not enough data, the cold start process will still finish, but then we can use the progress bar for displaying the estimated time left to finish initializing.

This is just an estimate, I figured it is better to provide some estimate rather than none at all.

Reply:
The cold start process will finish after 1 interval. But it may not succeed if there is no historical data. Then AD will rerun cold start in next interval until find some historical data and cold start run successfully.

I agree with you, as this is just an estimate. I see you will confirm with tech writer about the wording. Just make sure tech writer understand this case.

Reply:
So not sure I understand, are you saying that cold start will be re-ran every interval if the detector is still initializing then?

Reply:
If no historical data, cold start process ends but RCF model is not available. For this case, will rerun cold start in next interval and detector is still initializing.

Reply:
Oh, was not aware this was how this worked. I thought cold start was triggered once when the detector was first created, and not every interval where there wasn't sufficient historical data. This kind of defeats the entire purpose of adding this change then, since it will always show this callout until there is enough data, then go straight to running state.

Seems like the progress bar will never be shown. Am I understanding this correctly?

Reply:
My understanding 
1. If there is enough historical data for cold start and shingle find enough consecutive historical data points, then the progress bar show as 100% after cold start and shingle
2. If there is enough historical data for cold start and shingle doesn't find enough consecutive historical data points, then the progress bar show as X% after cold start, X=(shinlge size) / DETECTOR_SHINLGE_SIZE

@kaituo  can you confirm the cold start and shingle logic?

Reply:
My thoughts on this:

In case of `estimatedTime` unavailable, we may not need to add "This will take approximately X total minutes." Just say "Data is being collected to initialize model." in the intermediate callout. If `estimatedTime` is available, then show existing callout with progress bar.

Reply:
@ylwu-amzn 
X = completed shingles / required shingles

Reply:
> Oh, was not aware this was how this worked. I thought cold start was triggered once when the detector was first created, and not every interval where there wasn't sufficient historical data. This kind of defeats the entire purpose of adding this change then, since it will always show this callout until there is enough data, then go straight to running state.
> 
> Seems like the progress bar will never be shown. Am I understanding this correctly?
Your callout is necessary.  Cold start can finish without fully initializing the model.  What Yaliang said is the case there is no complete shingle in the history at all.  For example, if there is only one full shingle, cold start will finish with that.  We still need 127 shingles.  So the progress percentage is 1/128.

Reply:
> My thoughts on this:
> 
> In case of `estimatedTime` unavailable, we may not need to add "This will take approximately X total minutes." Just say "Data is being collected to initialize model." in the intermediate callout. If `estimatedTime` is available, then show existing callout with progress bar.

This can solve Yaliang's concern: cold start may not succeed.   Tyler said "approximately".  So it should cover cold start may not succeed case.  I am fine with both solutions: leave it as it is, or remove the cold start estimated finish time.

Reply:
Got it. That makes sense. I am fine for either soln as well. Could also change to give an estimation range, such as [interval - 6*interval], although this may be exposing too much of an internal implementation detail about the initialization process, and would have to be maintained.

I would like to assume in the majority of cases, the detector is detecting over an index with at least some data, in which case I think having an estimate could be much more helpful here.

Reply:
> I would like to assume in the majority of cases, the detector is detecting over an index with at least some data, in which case I think having an estimate could be much more helpful here.

Agree, should be an edge case that no historical data to do cold start.